src/msg/Messenger.h

```diff
@@ -406,6 +406,12 @@ class Messenger {
    * @return 0 on success, -errno otherwise.
    */
   virtual int shutdown() { started = false; return 0; }
+  /**
+   * Add the accepter to heartbeat check
+   * if accepter fails to receive any new connections, 
+   * suicide the osd thread.
```

Comment:
Do you mean this problem also exists in MON, MDS?

Reply:
no. i mean it's confusing to mention osd in this context. but instead of addressing the comments, i'd suggest just call `ceph_abort()` if we run into on unrecoverable error.

Reply:
I thought the whole reason to go through the heartbeat mechanism was that it will still shut down quickly but that shutdown will be a bit more graceful as it continues service for a bit and its peers can start timing it out.

If we think abort() is sufficient, it's obviously a lot faster...